src/main.cpp

```diff
@@ -203,32 +217,35 @@ int main(int argc, char** argv)
             auto showErrorAndQuit = [&app, file](const QString& errorMessage) {
                 QTextStream err(stderr);
                 err << errorMessage << Qt::endl;
-                app.exit(1);
+                app->exit(1);
             };
-            QObject::connect(window, &MainWindow::openFileError, &app, showErrorAndQuit);
-            QObject::connect(window, &MainWindow::exportFailed, &app, showErrorAndQuit);
-            QObject::connect(window, &MainWindow::exportFinished, &app, [&app, file](const QUrl& url) {
+            QObject::connect(&perfParser, &PerfParser::exportFailed, app.data(), showErrorAndQuit);
+            QObject::connect(&perfParser, &PerfParser::exportFinished, app.data(), [&app, file](const QUrl& url) {
                 QTextStream out(stdout);
                 out << QCoreApplication::translate("main", "Input file %1 exported to %2")
                            .arg(file, url.toDisplayString(QUrl::PrettyDecoded | QUrl::PreferLocalFile))
                     << Qt::endl;
-                app.exit(0);
+                app->exit(0);
             });
             auto destination = QUrl::fromUserInput(parser.value(exportTo), QDir::currentPath(), QUrl::AssumeLocalFile);
-            window->saveAs(file, destination);
-            return app.exec();
-        } else {
+            perfParser.startParseFile(file);
```

Comment:
...here this could do

```
auto perfParser = new PerfParser(app.data());
perfParser->startParseFile(file);
perfParser.exportResults(destination);
```

that said: is the order of the two commands even safe? what if parsing takes longer? what if it is finished already?